packages/client/vue.config.js

```diff
@@ -6,6 +6,12 @@ module.exports = {
   configureWebpack: {
     devtool: 'source-map',
   },
+  chainWebpack: (config) => {
+    config.plugin('copy').tap(([options]) => {
+      options.patterns[0].globOptions.ignore.push('**/deploy-config.js');
+      return [options];
+    });
+  },
```

Comment:
Adapted from [this StackOverflow answer](https://stackoverflow.com/a/60166838).

Would appreciate feedback (in either direction) from someone more familiar with webpack (and its Vue configuration) than I am... which is probably most people 🙃

Reply:
Can't say that I can offer another suggestion here. Does this work as intended locally and is there a test we can do when its deployed to `staging` to confirm if its working as we expect to in hosted environments as well?

Reply:
> Does this work as intended locally
It does!

> is there a test we can do when its deployed to staging to confirm if its working as we expect to in hosted environments as well?

Yes, once deployed to staging, there are two checks that can be done:
1. The site shouldn't be broken. Since this `deploy-config.js` file sets `apiURLForGOST = "/"`, it's basically relying on the Vue dev server's proxy to the API backend. Since that proxy doesn't exist in non-dev environments and since the API in Staging/Prod is hosted on a different subdomain, this version of `deploy-config.js` would break requests to the backend.
2. I'm 99.9% confident that the above won't happen, because our CloudFront CDN explicitly redirects requests to `/deploy-config.js` to the S3 object managed by Terraform. However, I'd still want to check that this file is absent in the S3 bucket path that webpack bundles get deployed to (which would confirm that this file is properly excluded from builds).

I'm honestly less concerned about the `deploy-config.js`-specific behavior here and more interested in making sure that there aren't any unintended consequences of this new configuration that may have slipped by me. My confidence is boosted because this snipped of code seems fairly ubiquitous in StackOverflow, GitHub issues, and the Vue support forum... so at least we're in good company! 😅